channeldb/invoices.go

```diff
@@ -314,16 +314,22 @@ type HtlcAcceptDesc struct {
 // invoice.
 type InvoiceUpdateDesc struct {
 	// State is the new state that this invoice should progress to.
-	State ContractState
+	State *InvoiceStateUpdateDesc
 
 	// CancelHtlcs describes the htlcs that need to be canceled.
 	CancelHtlcs map[CircuitKey]struct{}
 
 	// AddHtlcs describes the newly accepted htlcs that need to be added to
 	// the invoice.
 	AddHtlcs map[CircuitKey]*HtlcAcceptDesc
+}
+
+// InvoiceStateUpdateDesc describes an invoice-level state transition.
+type InvoiceStateUpdateDesc struct {
```

Comment:
I don't really see why these two fields have to go into a new struct

Reply:
To allow the `StateUpdate` to be nil, meaning no state update.

Reply:
Yes, it is a bit more strict like this. Also a matter of personal pref.